src/providers/be_dyndns.c

```diff
@@ -1126,16 +1139,30 @@ struct tevent_req *be_nsupdate_send(TALLOC_CTX *mem_ctx,
 static const char **
 be_nsupdate_args(TALLOC_CTX *mem_ctx,
                  enum be_nsupdate_auth auth_type,
-                 bool force_tcp)
+                 bool force_tcp,
+                 struct sss_parsed_uri *server_uri,
+                 const char *dot_cacert,
+                 const char *dot_cert,
+                 const char *dot_key)
 {
     const char **argv;
     int argc = 0;
+    bool use_dot;
+    bool have_dot_cert;
+    bool have_dot_key;
 
-    argv = talloc_zero_array(mem_ctx, const char *, 6);
+    argv = talloc_zero_array(mem_ctx, const char *, 14);
     if (argv == NULL) {
         return NULL;
     }
 
+    if (! sss_is_valid_dns_scheme(server_uri->scheme)) {
+        DEBUG(SSSDBG_IMPORTANT_INFO, "Invalid DNS scheme: %s, using dns://\n", server_uri->scheme);
```

Comment:
Maybe add syslog message if you choose to continue?

Reply:
Moved to sss_log.

Reply:
Can you also keep DEBUG? We usually log to both.

Reply:
This still stands.